test/jdk/javax/swing/JTable/bug4190222.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 1998, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.Robot;
+import java.util.Vector;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.DefaultTableModel;
+
+/*
+ * @test
+ * @bug 4190222
+ * @summary Setting data vector on the model correctly repaint table
+ * @key headful
+ * @run main bug4190222
+ */
+
+public class bug4190222 {
+    static JFrame frame;
+    static DefaultTableModel dtm;
+    static JTable tbl;
+    static volatile Dimension preResize;
+    static volatile Dimension postResize;
+
+    static Vector data;
+    static Vector colNames;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(250);
+
+            SwingUtilities.invokeAndWait(() -> createTestUI());
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() -> {
+                preResize = tbl.getSize();
```

Comment:
If you do all the sizes retrieval inside the single invokeAndWait block you can also do a comparison here and throw an exception if sizes mismatch. This way you only need one variable preResize and you can do comparison as
`if (!preResize.equals(tbl.getSize()) ...` so you will only need only one local variable. It is not critical but looks cleaner.